packages/components/src/internal/components/labels/LabelsConfigurationPanel.spec.tsx

```diff
@@ -0,0 +1,179 @@
+        }),
+        defaultLabel: undefined,
+    };
+
+    test('default props', () => {
+        const wrapper = mountWithAppServerContext(<LabelsConfigurationPanel {...DEFAULT_PROPS} />);
+
+        expect(wrapper.find(LabelTemplatesList)).toHaveLength(1);
+        expect(wrapper.find(LabelTemplateDetails)).toHaveLength(1);
+    });
+});
+
+describe('LabelTemplatesList', () => {
+    const DEFAULT_PROPS = {
+        onSelect: jest.fn(),
+        selected: undefined,
+        templates: [],
+    };
+
+    test('default props', () => {
+        const wrapper = mountWithAppServerContext(<LabelTemplatesList {...DEFAULT_PROPS} />);
+
+        expect(wrapper.find('.choices-list__empty-message')).toHaveLength(1);
+        expect(wrapper.find(ChoicesListItem)).toHaveLength(0);
+    });
+
+    test('Single Item', () => {
+        const wrapper = mountWithAppServerContext(
+            <LabelTemplatesList
+                {...DEFAULT_PROPS}
+                templates={[
+                    new LabelTemplate({
+                        name: 'T1',
+                        path: 'T1_path',
+                        rowId: 0,
+                    }),
+                ]}
+            />
+        );
+
+        expect(wrapper.find('.choices-list__empty-message')).toHaveLength(0);
+        expect(wrapper.find(ChoicesListItem)).toHaveLength(1);
+    });
+
+    test('Two Items', () => {
+        const wrapper = mountWithAppServerContext(
+            <LabelTemplatesList
+                {...DEFAULT_PROPS}
+                templates={[
+                    new LabelTemplate({
+                        name: 'T1',
+                        path: 'T1_path',
+                        rowId: 0,
+                    }),
+                    new LabelTemplate({
+                        name: 'T2',
+                        path: 'T2_path',
+                        rowId: 1,
+                    }),
+                ]}
+            />
+        );
+
+        expect(wrapper.find('.choices-list__empty-message')).toHaveLength(0);
+        expect(wrapper.find(ChoicesListItem)).toHaveLength(2);
+    });
+
+    test('Two Items, with default', () => {
+        const wrapper = mountWithAppServerContext(
+            <LabelTemplatesList
+                {...DEFAULT_PROPS}
+                defaultLabel={0}
+                templates={[
+                    new LabelTemplate({
+                        name: 'T1',
+                        path: 'T1_path',
+                        rowId: 0,
+                    }),
+                    new LabelTemplate({
+                        name: 'T2',
+                        path: 'T2_path',
+                        rowId: 1,
+                    }),
+                ]}
+            />
+        );
+
+        expect(wrapper.find('.choices-list__empty-message')).toHaveLength(0);
+        expect(wrapper.find(ChoicesListItem)).toHaveLength(2);
+        expect(wrapper.find('.badge')).toHaveLength(1);
```

Comment:
should check that this `.badge` item doesn't exist for the cases above where default isn't set

Reply:
added